core/app/models/spree/shipping_rate.rb

```diff
@@ -1,46 +1,48 @@
 module Spree
   class ShippingRate < Spree::Base
+    attr_writer :order
```

Comment:
Yes, that's exactly the reason. The estimator doesn't really build shipments at all, one of the other `stock` classes does that. And no, a shipping rate without a shipment doesn't really make a lot of sense. But I (honestly) get a little lost between packages, inventory units and shipments at times, so I opted for not refactoring that here. If anyone has a good idea on how to avoid this, I'm all for it.